File: output/deepseek_deepseek_v3_2/zero/python/Fita Colorida.py

import sys

def main():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    n = int(data[0])
    fita = list(map(int, data[1:1+n]))
    
    INF = 10**9
    dist = [INF] * n
    
    # Passo da esquerda para a direita
    last_zero = -INF
    for i in range(n):
        if fita[i] == 0:
            last_zero = i
        dist[i] = min(dist[i], i - last_zero)
    
    # Passo da direita para a esquerda
    last_zero = INF
    for i in range(n-1, -1, -1):
        if fita[i] == 0:
            last_zero = i
        dist[i] = min(dist[i], last_zero - i)
    
    # Colorir a fita
    for i in range(n):
        if fita[i] == 0:
            fita[i] = 0
        else:
            d = dist[i]
            if d >= 9:
                fita[i] = 9
            else:
                fita[i] = d
    
    print(' '.join(map(str, fita)))

if __name__ == "__main__":
    main()

File: output/stepfun_step_3_5_flash_free/zero/cpp/Fita Colorida.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <climits>

using namespace std;

int main() {
    int N;
    cin >> N;
    vector<int> fita(N);
    for (int i = 0; i < N; i++) {
        cin >> fita[i];
    }

    const int INF = 1e9;
    vector<int> dist(N, INF);

    for (int i = 0; i < N; i++) {
        if (fita[i] == 0) {
            dist[i] = 0;
        }
    }

    for (int i = 1; i < N; i++) {
        if (dist[i-1] != INF) {
            dist[i] = min(dist[i], dist[i-1] + 1);
        }
    }

    for (int i = N-2; i >= 0; i--) {
        if (dist[i+1] != INF) {
            dist[i] = min(dist[i], dist[i+1] + 1);
        }
    }

    for (int i = 0; i < N; i++) {
        if (dist[i] >= 9) {
            dist[i] = 9;
        }
    }

    for (int i = 0; i < N; i++) {
        if (i > 0) cout << " ";
        cout << dist[i];
    }
    cout << endl;

    return 0;
}